rs/xnet/payload_builder/src/lib.rs

```diff
@@ -231,6 +231,9 @@ impl XNetPayloadBuilderMetrics {
     }
 }
 
+// TODO: Unify this constant and `MAX_STREAM_MESSAGES` in the stream builder.
+pub const MAX_STREAM_MESSAGES: usize = 10_000;
```

Comment:
Probably not for this MR, but we should either make the constant in `StreamBuilder` pub or else define the constant somewhere else or perhaps even better make this an argument passed in from the outside.

This way we could make it part of some config and thus configurable in state machine tests. It would be great if we could put these limits to something much smaller for testing. The stream size and also this change here won't trigger a crash or anything of the sort of the constant is changed as far as I know.

Reply:
Actually even for this change it would be useful to have it configurable in proptests so that we don't have to generate a stream with > 10k messages in it every iteration.

Reply:
Agreed about making this configurable, either only on paper (by having a constant somewhere in `/rs/replica` and passing it in the constructors of all relevant components); or by actually adding a field somewhere under `rs/config` (and passing it the same way).

In either case (but especially in the latter) we must be careful with how we validate that this limit is correctly enforced.

Reply:
Made a [ticket](https://dfinity.atlassian.net/jira/software/c/projects/MR/boards/223?selectedIssue=MR-636).